test/image/mocks/zz-text_on_shapes_texttemplate.json

```diff
@@ -0,0 +1,334 @@
+                    "yanchor": "top"
+                },
+                "layer": "above",
+                "xref": "x3",
+                "yref": "y3",
+                "x0": 1,
+                "x1": 10000,
+                "y0": 1000000,
+                "y1": 10000,
+                "type": "line",
+                "opacity": 0.5,
+                "line": {
+                    "width": 3,
+                    "color": "#025"
+                },
+                "editable": true
+            },
+            {
+                "label": {
+                    "texttemplate": "x0: %{x0:.2f}<br>x1: %{x1:.2f}<br>xcenter: %{xcenter:.2f}<br>dx: %{dx:.2f}<br>width: %{width:.2f}<br><br>y0: %{y0:.2f}<br>y1: %{y1:.2f}<br>ycenter: %{ycenter:.2f}<br>dy: %{dy:.2f}<br>height: %{height:.2f}<br><br>length: %{length}"
+                },
+                "layer": "above",
+                "xref": "x3",
+                "yref": "y3",
+                "x0": 10,
+                "x1": 1000,
+                "y0": 0.25,
+                "y1": 20000,
+                "type": "rect",
+                "line": {
+                    "width": 2,
+                    "color": "#025"
+                },
+                "editable": true
+            },
+            {
+                "label": {
+                    "texttemplate": "Rate = %{slope*31536000000:.3f} per year"
+                },
+                "layer": "above",
+                "xref": "x4",
+                "yref": "y4",
+                "x0": "2021-01-01",
+                "x1": "2023-01-01",
+                "y0": 2.5,
+                "y1": 6,
+                "type": "line",
+                "line": {
+                    "width": 2,
+                    "color": "#025"
+                },
+                "editable": true
+            },
+            {
+                "label": {
+                    "texttemplate": "x0: %{x0|%m %b %Y}<br>x1: %{x1|%m %b %Y}<br>xcenter: %{xcenter|%m %b %Y}<br>dx: %{dx/86400000:d} days<br>width: %{width/86400000:d} days<br><br>y0: %{y0:.1f}<br>y1: %{y1:.1f}<br>ycenter: %{ycenter:.1f}<br>dy: %{dy:.1f}<br>height: %{height:.1f}<br>",
+                    "textposition": "top left"
+                },
+                "layer": "above",
+                "xref": "x4",
+                "yref": "y4",
+                "x0": "2021-07-01",
+                "x1": "2022-07-01",
+                "y0": 2.5,
+                "y1": 10,
+                "type": "rect",
+                "line": {
+                    "width": 2,
+                    "color": "#025"
+                },
+                "editable": true
+            }
+        ],
+        "newshape": {
+            "label": {
+                "texttemplate": "Slope! %{slope*31536000000:.2f}"
+            }
+        }
+    },
+    "config": {
+        "modeBarButtonsToAdd": [
+            "drawline",
+            "drawopenpath",
+            "drawclosedpath",
+            "drawcircle",
+            "drawrect",
```

Comment:
This is a beautiful mock!
BTW I noticed if one selects `drawrect` or `drawcircle` and then start drawing new shapes; it prints out some `slope` values which it shouldn't.
So we need to fix that.

Reply:
yeah initially I was thinking we should just allow it, but the more I think about it, the more it seems like we should not allow `slope` for `rect` and `circle`, it just seems kind of weird.